src/helm/proxy/clients/huggingface_model_registry.py

```diff
@@ -71,36 +76,54 @@ def from_string(raw: str) -> "HuggingFaceModelConfig":
             namespace=match.group("namespace"), model_name=model_name, revision=match.group("revision")
         )
 
+    @staticmethod
+    def from_path(path: str) -> "HuggingFaceModelConfig":
+        """Genertes a HuggingFaceModelConfig from a (relative or absolute) path to a local HuggingFace model."""
+        pattern = r"((?P<containing_dir>.+?)/)?(?P<model_name>[^/@]+)(@(?P<revision>[^/@]+))?/?"
+        match = re.fullmatch(pattern, path)
+        if not match:
+            raise ValueError(f"Could not parse model path: '{path}'; Expected format: /path/to/model_name[@revision]")
+        model_name = match.group("model_name")
+        assert model_name
+        return HuggingFaceModelConfig(
+            namespace="local",
```

Comment:
Setting namespace to "local" is dubious because there's actually a namespace called "local" on Hugging Face Hub: https://huggingface.co/local

Let's just make the namespace `None`. It's still ambiguous because it could refer to a hub model with no namespace like `gpt2`. But it's more obvious that it's intended to be a sentinel null-ish value.

Reply:
Why don't we put something more explicit? I personally feel like "None" is not super clear. Could we maybe put something like "internal"? "local-model"?